### ISSUE TITLE
Problem loading dataset for training

### ISSUE THREAD

**Issue by collinabidi** [dgominski/generalizablersc#1]
Hi! Great work, your paper was very interesting to read at CVPR, congratulations on Best Paper for Earthvision!

After downloading the datasets, setting the DATAROOT of each dataset class accordingly, and running `train.py`, I'm seeing an error regarding properly loading the UC Merced data. I've attached a picture that suggests that the class folder `data/datasets/UCMerced_LandUse/Images/freeway` is being considered a part of the dataset.

![error-screenshot](https://user-images.githubusercontent.com/29078081/182894013-c1f3ef3b-248e-4030-a00b-9b71220c2924.png)

My DATAROOT in `ucm_dataset.py` is as follows: `self.dataroot = "./data/datasets/UCMerced_LandUse/Images"`. The picture below shows this. The location of the dataset can be seen in the sidebar.

![ucm_configuration](https://user-images.githubusercontent.com/29078081/182895301-f850b3ab-38de-4327-83bc-d8c831b3a646.png)

Edit: I've confirmed that the class folders are being considered images, as the `paths.npy` file generated in Line 155 and saved in Line 176 of `multi_dataset.py` contains folder paths, as seen in the image below. Is this behavior intentional?

![paths](https://user-images.githubusercontent.com/29078081/182900191-24eda8b2-8fb1-4e8e-b0d8-54b850d7cc7b.png)

Thank you,
Collin

**Comment by dgominski** [comment]
Hello Collin,
Thanks for you interest !
I cannot reproduce this behavior on my local project, in your case glob seems to grab folder names somehow.
Could you try replacing line 50 above in ucm_dataset.py:
              `  tempdf = pd.DataFrame({'classname': os.path.basename(classname), 'class': i, 'path': glob.glob(os.path.join(classname, "*"))})`
with:
              `  tempdf = pd.DataFrame({'classname': os.path.basename(classname), 'class': i, 'path': glob.glob(os.path.join(classname, "*.tif"))})`

and tell me if that solves the problem for UCM ?
If that does solve the problem I will add the extension condition to all datasets to make sure only images are grabbed.

**Comment by collinabidi** [comment]
That worked, thank you! Adding the same fix to all other classes (with their appropriate extension) fixed the same issue in other dataset class files. 

I would also suggest perhaps adding a `--use_cache` option to the command-line arguments that enables/disables the use of `data.pkl` and `queries.pkl`. Since I had to regenerate the correct paths, I manually removed all of the `data.pkl` and `queries.pkl` files from each dataset directory. Not a massive hassle, but it might be a nice feature :).
